### ISSUE TITLE
 Seem not working in android. I get "Illegal audio focus gain type 0" on player.resume();

### ISSUE THREAD

**Issue by maxguy** [bluefireteam/audioplayers#1301]
## Checklist
- [x] I read the [troubleshooting guide](https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md) before raising this issue
- [x] I made sure that the issue I am raising doesn't already exist

## Current bug behaviour

Edit(Gustl22): Setting `audioFocus: AndroidAudioFocus.none`

## Steps to reproduce

1. Execute `flutter run` on the code sample 
2. ... 
3. ... 

<details>
<summary>Code sample</summary>

```dart
final AudioContext audioContext = AudioContext(
        iOS: AudioContextIOS(
          defaultToSpeaker: true,
          category: AVAudioSessionCategory.ambient,
          options: [
            AVAudioSessionOptions.defaultToSpeaker,
            AVAudioSessionOptions.mixWithOthers,
          ],
        ),
        android: AudioContextAndroid(
          isSpeakerphoneOn: true,
          stayAwake: true,
          contentType: AndroidContentType.sonification,
          usageType: AndroidUsageType.assistanceSonification,
          audioFocus: AndroidAudioFocus.none,
        ),
      );
      AudioPlayer.global.setGlobalAudioContext(audioContext);
```

```dart
void main() {
}
```

</details>

## Logs

```
my relevant logs
```

<details>
  <summary>Full Logs</summary>

  

  ```
  my full logs or a link to a gist
  ```

  Flutter doctor:
  ```
  Output of: flutter doctor -v
  ```
</details>

## Audio Files/URLs/Sources

- Sample Url: https://luan.xyz/files/audio/coins.wav
- Sample Stream: https://example.com/my_stream.m3u8

## Environment information

* audioplayers version: Edit(Gustl22): 2.0.0

Platform 1: 
* OS name and version: 
* Device: 

* build mode: `debug`

* error peculiarities (optional): 

Platforms tested without any issue (optional): 
* test peculiarities:

**Comment by alexcherniatin** [comment]
Problem caused by setting audio context with: "audioFocus: AndroidAudioFocus.none" value.

**Comment by rsyd29** [comment]
> AndroidAudioFocus.none

any update?

**Comment by Gustl22** [comment]
@rsyd29 @alexcherniatin which AudioPlayers version do you use?

**Comment by alexcherniatin** [comment]
Audioplayers version: "2.0.0".

But I don't know why somebody needs AndroidAudioFocus.none, with any other option, all working as expected.

**Comment by wujek-srujek** [comment]
I did a lot of testing in the last couple of days and all I can say is that `AndroidAudioFocus.none` simply isn't a value you should be using, it always causes a crash, not sure what it is for, I guess it is available because Android has it when no focus operations are requested. However, if you are using an `AudioContext` for whatever reason, there is no way to revert the Android focus behavior to the default, see https://github.com/bluefireteam/audioplayers/issues/1495 for more details.

**Comment by Gustl22** [comment]
Fixed in #1534

**Comment by Jonny5-5** [comment]
> I did a lot of testing in the last couple of days and all I can say is that `AndroidAudioFocus.none` simply isn't a value you should be using, it always causes a crash, not sure what it is for, I guess it is available because Android has it when no focus operations are requested. However, if you are using an `AudioContext` for whatever reason, there is no way to revert the Android focus behavior to the default, see #1495 for more details.

That's because AndroidAudioFocus.none means that Android won't give your application focus to play any audio.  Set it to another value and it should work fine.
